jax/_src/scipy/spatial/transform.py

```diff
@@ -0,0 +1,440 @@
+
+  quat: jax.Array
+
+  @classmethod
+  def concatenate(cls, rotations: typing.Sequence):
+    """Concatenate a sequence of `Rotation` objects."""
+    return cls(jnp.concatenate([rotation.quat for rotation in rotations]))
+
+  @classmethod
+  def from_euler(cls, seq: str, angles: jax.Array, degrees: bool = False):
+    """Initialize from Euler angles."""
+    assert angles.ndim in [1, 2]
+    num_axes = len(seq)
+    if num_axes < 1 or num_axes > 3:
+      raise ValueError("Expected axis specification to be a non-empty "
+                       "string of upto 3 characters, got {}".format(seq))
+    intrinsic = (re.match(r'^[XYZ]{1,3}$', seq) is not None)
+    extrinsic = (re.match(r'^[xyz]{1,3}$', seq) is not None)
+    if not (intrinsic or extrinsic):
+      raise ValueError("Expected axes from `seq` to be from ['x', 'y', "
+                       "'z'] or ['X', 'Y', 'Z'], got {}".format(seq))
+    if any(seq[i] == seq[i+1] for i in range(num_axes - 1)):
+      raise ValueError("Expected consecutive axes to be different, "
+                       "got {}".format(seq))
+    seq = seq.lower()
+    _from_euler = lambda a: _elementary_quat_compose(a, seq, intrinsic, degrees)
+    if angles.ndim == 1:
+      return cls(_from_euler(angles))
+    else:
+      return cls(jax.vmap(_from_euler)(angles))
+
+  @classmethod
+  def from_matrix(cls, matrix: jax.Array):
+    """Initialize from rotation matrix."""
+    assert matrix.ndim in [2, 3]
+    if matrix.ndim == 2:
+      return cls(_from_matrix(matrix))
+    else:
+      return cls(jax.vmap(_from_matrix)(matrix))
+
+  @classmethod
+  def from_mrp(cls, mrp: jax.Array):
+    """Initialize from Modified Rodrigues Parameters (MRPs)."""
+    assert mrp.ndim in [1, 2]
+    if mrp.ndim == 1:
+      return cls(_from_mrp(mrp))
+    else:
+      return cls(jax.vmap(_from_mrp)(mrp))
+
+  @classmethod
+  def from_quat(cls, quat: jax.Array):
+    """Initialize from quaternions."""
+    assert quat.ndim in [1, 2]
+    if quat.ndim == 1:
+      return cls(_normalize_quaternion(quat))
+    else:
+      return cls(jax.vmap(_normalize_quaternion)(quat))
+
+  @classmethod
+  def from_rotvec(cls, rotvec: jax.Array, degrees: bool = False):
+    """Initialize from rotation vectors."""
+    assert rotvec.ndim in [1, 2]
+    if rotvec.ndim == 1:
+      return cls(_from_rotvec(rotvec, degrees))
+    else:
+      return cls(jax.vmap(_from_rotvec, in_axes=[0, None])(rotvec, degrees))
+
+  @classmethod
+  def identity(cls, num: typing.Optional[int] = None):
+    """Get identity rotation(s)."""
+    assert num is None
+    quat = jnp.array([0, 0, 0, 1])
+    return cls(quat)
+
+  # @classmethod
+  # def random(cls, random_key: jax.random.PRNGKey, num: typing.Optional[int] = None):
+  #   """Generate uniformly distributed rotations."""
+  #   # Need to implement scipy.stats.special_ortho_group for this to work...
+
+  def __bool__(self):
+    """Comply with Python convention for objects to be True."""
+    return True
```

Comment:
Is this necessary? This is a non-empty NamedTuple so `__bool__` will default to `True` I think?

Reply:
Sounds good, removed this function.